libsolidity/ast/Types.cpp

```diff
@@ -596,17 +596,23 @@ TypePointer FixedPointType::binaryOperatorResult(Token::Value _operator, TypePoi
 		return commonType;
 	if (Token::isBitOp(_operator) || Token::isBooleanOp(_operator))
 		return TypePointer();
-	if (auto fixType = dynamic_pointer_cast<FixedPointType const>(commonType))
+	if (_operator == Token::Exp)
```

Comment:
Please disallow exponentiation for now. I think it should return the base type and not the common type to be consistent with `<<`, but I'm not sure. What do you think, @axic?

Reply:
I think it should be consistent with how it works on integer types. That works on the common type. I don't actually see a good reason right now to do exponentiation on fixed point types, so would keep it disabled, and reconsider when the need arises.

Reply:
Also, integer types disallow exponentation with fixed point types, at the bare minimum the same restriction should apply here.